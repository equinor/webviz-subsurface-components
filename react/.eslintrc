{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends":  [
    "plugin:react/recommended",
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "plugins": ["@typescript-eslint"],
      "extends": [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended"
      ]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "amd": true,
    "browser": true,
    "es6": true,
    "jasmine": true,
    "node": true
  }
}
