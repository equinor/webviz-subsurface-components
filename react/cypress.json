{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true
}
