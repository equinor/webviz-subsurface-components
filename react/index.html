<html>
    <head>
        <title>Demo application of webviz-subsurface-components</title>
    </head>
    <body>
        <div id='root'></div>
        <link rel="stylesheet" type="text/css" href="output.css">
        <script src="output.js"></script>
    </body>
</html>
