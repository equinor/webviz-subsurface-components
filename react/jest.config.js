/**
 * The jest tests had an error: SyntaxError: Cannot use import statement outside a module
 * After some investigation, it leads to this page https://github.com/facebook/jest/issues/9395.
 */
module.exports = {
    moduleNameMapper: {
        "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
            "<rootDir>/__mocks__/fileMock.js",
        "\\.(css|less)$": "<rootDir>/__mocks__/styleMock.js",
    },
    transform: {
        "\\.(js|ts|jsx|tsx)$": [
            "babel-jest",
            { configFile: "./config/babel.config.json" },
        ],
    },
    transformIgnorePatterns: ["<rootDir>/node_modules/(?!(@webviz)/)"],
};
