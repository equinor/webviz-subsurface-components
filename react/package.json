{
    "name": "@webviz/subsurface-components",
    "description": "Custom Dash components for use in Webviz",
    "repository": {
        "type": "git",
        "url": "git://github.com/equinor/webviz-subsurface-components.git"
    },
    "bugs": {
        "url": "https://github.com/equinor/webviz-subsurface-components/issues"
    },
    "homepage": "https://github.com/equinor/webviz-subsurface-components",
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/**/*"
    ],
    "scripts": {
        "start": "webpack serve --mode development --devtool inline-source-map --entry ./src/demo/index.js --open",
        "copy-package-json": "copyfiles ./package.json ../webviz_subsurface_components/",
        "transpile": "tsc --project tsconfig.build.json && npm run copy-files",
        "transpile:dash": "tsc --project tsconfig.build.json && rimraf ./dist/components/*/**/",
        "tidy-up": "rimraf ./dist ./webviz_subsurface_components",
        "build:js": "NODE_ENV=production webpack --mode production",
        "build:js-dev": "NODE_ENV=development webpack --mode development",
        "build:js-demo": "NODE_ENV=production webpack --mode production --entry ./src/demo/index.js",
        "build:py": "npm run transpile:dash && copyfiles -a ../webviz_subsurface_components/. ./webviz_subsurface_components/ && copyfiles ./package.json ./webviz_subsurface_components/ && copyfiles -f ./src/lib/assets/VectorDefinitions.json ./webviz_subsurface_components/ && dash-generate-components ./dist/components webviz_subsurface_components -p package.json --ignore '(^index.js|.stories.js)$' && copyfiles -a -f ./webviz_subsurface_components/* ../webviz_subsurface_components/",
        "build": "npm run transpile && npm run build:js && npm run build:js-dev && npm run tidy-up && npm run build:py && npm run tidy-up",
        "typecheck": "tsc --noEmit",
        "format": "eslint --fix *.js *json \"src/**/*.+(ts|tsx|js|jsx|json|css)\"",
        "lint": "eslint *.js *.json \"src/**/*.+(ts|tsx|js|jsx|json|css)\"",
        "validate": "npm run typecheck && npm run lint",
        "setup-deckgl-types": "cd ./node_modules/@danmarshall/deckgl-typings/ && indefinitely-typed --folder deck.gl__aggregation-layers --folder deck.gl__core --folder deck.gl__extensions --folder deck.gl__geo-layers --folder deck.gl__google-maps --folder deck.gl__json --folder deck.gl__layers --folder deck.gl__mapbox --folder deck.gl__mesh-layers --folder deck.gl__react --folder deck.gl --folder luma.gl__constants --folder luma.gl__core --folder luma.gl__gltools --folder luma.gl__webgl-state-tracker --folder luma.gl__webgl --folder math.gl__core --folder math.gl",
        "storybook": "start-storybook -p 6006",
        "build-storybook": "build-storybook",
        "test_perf": "jest _performance",
        "test_correctness": "jest --coverage --testPathIgnorePatterns='_performance'",
        "test": "jest --coverage",
        "test:watch": "npm test -- --watch",
        "build:npm": "rimraf dist && tsc -project tsconfig.build.json --declaration && npm run copy-files",
        "copy-files": "copyfiles package.json dist && copyfiles --up 1 ../README.md dist && copyfiles --exclude **/*.{tsx,jsx,js,ts,snap} --up 2 \"src/lib/**/*\" dist/",
        "prepare": "npm run build:npm",
        "postpack": "npm run tidy-up",
        "prepublishOnly": "npm test && npm run lint",
        "preversion": "npm run lint",
        "test-cypress": "concurrently 'npm run storybook' 'npm run cypress:test' -k -s first",
        "cypress:test": "wait-on http://localhost:6006 && cypress run",
        "cypress:run": "cypress run"
    },
    "author": "Equinor <opensource@equinor.com>",
    "license": "MPL",
    "dependencies": {
        "@emerson-eps/color-tables": "^0.4.22",
        "@equinor/eds-core-react": "^0.12.1",
        "@equinor/eds-icons": "^0.9.1",
        "@equinor/videx-wellog": "^0.7.0",
        "@material-ui/core": "^4.12.3",
        "@material-ui/lab": "^4.0.0-alpha.60",
        "@react-leaflet/core": "^1.1.1",
        "@reduxjs/toolkit": "^1.7.2",
        "@types/d3": "^6.2.0",
        "@types/geojson": "^7946.0.7",
        "@types/react": "^17.0.3",
        "@types/react-dom": "^17.0.3",
        "@types/react-redux": "^7.1.15",
        "@types/react-resize-detector": "^5.0.0",
        "@ungap/structured-clone": "^0.3.4",
        "@vivaxy/png": "^1.3.0",
        "@webviz/core-components": "^0.6.0",
        "ajv": "^7.2.1",
        "clsx": "^1.1.1",
        "convert-units": "^2.3.4",
        "d3": "^5.11.0",
        "d3-format": "^1.4.5",
        "deck.gl": "^8.8.11",
        "deep-equal": "^2.0.5",
        "fast-json-patch": "^3.0.0-1",
        "jsonschema": "^1.4.0",
        "jsverify": "^0.8.4",
        "leaflet": "^1.6.0",
        "leaflet-draw": "^1.0.4",
        "lodash": "^4.17.21",
        "mathjs": "^9.4.2",
        "nebula.gl": "^0.22.3",
        "react": "^17.0.2",
        "react-dom": "^17.0.2",
        "react-dropdown-tree-select": "^2.5.1",
        "react-leaflet": "2.8",
        "react-leaflet-draw": "^0.19.8",
        "react-redux": "^7.2.6",
        "react-resize-detector": "^7.0.0",
        "react-tooltip": "^4.2.21",
        "redux": "^4.0.5",
        "semver": "^7.3.5",
        "styled-components": "^5.3.3",
        "uuid": "^8.3.2"
    },
    "devDependencies": {
        "@actions/core": "^1.9.1",
        "@babel/core": "^7.13.10",
        "@babel/plugin-proposal-class-properties": "^7.13.0",
        "@babel/preset-env": "^7.13.12",
        "@babel/preset-react": "^7.12.13",
        "@babel/preset-typescript": "^7.13.0",
        "@babel/runtime": "^7.13.10",
        "@storybook/addon-docs": "^6.5.9",
        "@storybook/addon-essentials": "^6.5.9",
        "@storybook/addon-links": "^6.5.9",
        "@storybook/addon-storysource": "^6.5.9",
        "@storybook/builder-webpack5": "^6.5.9",
        "@storybook/manager-webpack5": "^6.5.9",
        "@storybook/react": "^6.5.9",
        "@testing-library/cypress": "^8.0.2",
        "@testing-library/jest-dom": "^5.11.9",
        "@testing-library/react": "^12.1.2",
        "@testing-library/user-event": "^13.1.9",
        "@types/cypress-image-snapshot": "^3.1.6",
        "@types/d3-color": "^3.0.2",
        "@types/d3-interpolate": "^3.0.1",
        "@types/jest": "^26.0.20",
        "@types/lodash": "^4.14.171",
        "@types/redux-mock-store": "^1.0.3",
        "@types/semver": "^7.3.8",
        "@types/ungap__structured-clone": "^0.3.0",
        "@types/uuid": "^8.3.0",
        "@typescript-eslint/eslint-plugin": "^4.19.0",
        "@typescript-eslint/parser": "^4.19.0",
        "ace-builds": "^1.4.12",
        "addon-redux": "^2.0.11",
        "babel-jest": "^26.6.3",
        "babel-loader": "^8.1.0",
        "concurrently": "^7.0.0",
        "copyfiles": "^2.0.0",
        "core-js": "^3.9.1",
        "css-loader": "^5.2.0",
        "css-minimizer-webpack-plugin": "^3.0.1",
        "cssnano": "^5.0.5",
        "cypress": "^9.4.1",
        "d3-color": "^3.1.0",
        "dotenv-webpack": "^7.0.2",
        "eslint": "^7.0.0",
        "eslint-config-prettier": "^8.1.0",
        "eslint-plugin-import": "^2.18.2",
        "eslint-plugin-prettier": "^3.1.3",
        "eslint-plugin-react": "^7.19.0",
        "file-loader": "^6.2.0",
        "hex-and-rgba": "^2.0.0",
        "html-webpack-plugin": "^5.3.1",
        "indefinitely-typed": "^1.1.0",
        "jest": "^26.6.3",
        "jest-styled-components": "^7.0.8",
        "json-schema": "^0.4.0",
        "mini-css-extract-plugin": "^1.3.9",
        "node-sass": "^7.0.3",
        "postcss": "^8.3.0",
        "prettier": "^2.5.1",
        "process": "^0.11.10",
        "raw-loader": "^4.0.2",
        "react-ace": "^9.5.0",
        "react-docgen": "^5.3.1",
        "react-test-renderer": "^17.0.2",
        "redux-mock-store": "^1.5.4",
        "sass": "^1.36.0",
        "sass-loader": "^12.1.0",
        "source-map-loader": "^3.0.0",
        "storybook-addon-performance": "^0.16.1",
        "terser-webpack-plugin": "^5.1.1",
        "trim": ">=0.0.3",
        "ts-loader": "^9.2.6",
        "typescript": "4.3.2",
        "url-loader": "^4.1.1",
        "vue-style-loader": "^4.1.3",
        "wait-on": "^6.0.1",
        "webpack": "^5.70.0",
        "webpack-cli": "^4.5.0",
        "webpack-conditional-loader": "^1.0.12",
        "webpack-dev-server": "^4.11.0"
    },
    "engines": {
        "node": ">=14.17",
        "npm": ">=6.14"
    }
}
