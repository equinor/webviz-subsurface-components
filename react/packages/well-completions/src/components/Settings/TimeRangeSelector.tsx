import { NativeSelect } from "@equinor/eds-core-react";
import { styled } from "@mui/material/styles";
import { Slider } from "@mui/material";
import { isEqual } from "lodash";
import React, { useCallback, useContext, useMemo } from "react";
import { useDispatch, useSelector } from "react-redux";
import { updateTimeIndexRange } from "../../redux/actions";
import { WellCompletionsState } from "../../redux/store";
import { DataContext } from "../DataLoader";
const PREFIX = "TimeRangeSelector";

const classes = {
    root: `${PREFIX}-root`,
    valueLabel: `${PREFIX}-valueLabel`,
    slider: `${PREFIX}-slider`,
    selector: `${PREFIX}-selector`,
};

const Root = styled("div")(({ theme }) => ({
    [`& .${classes.root}`]: {
        display: "flex",
        flexDirection: "row",
    },

    [`& .${classes.slider}`]: {
        width: "200px",
        marginLeft: theme.spacing(4),
        marginRight: theme.spacing(4),
    },

    [`& .${classes.selector}`]: {
        maxWidth: "130px",
    },
}));

const EdsSlider = styled(Slider)({
    root: {
        color: "#007079",
    },
    valueLabel: {
        top: 22,
        "& *": {
            background: "transparent",
            color: "#000",
        },
    },
});

/**
 * A React component for selecting time step(s) to display in the plot
 */
const TimeRangeSelector: React.FC = React.memo(() => {
    // Direct access to the input data
    const data = useContext(DataContext);

    // Redux
    const dispatch = useDispatch();
    const timeAggregation = useSelector(
        (state: WellCompletionsState) => state.ui.timeAggregation
    );
    const timeIndexRange = useSelector(
        (state: WellCompletionsState) => state.ui.timeIndexRange,
        isEqual
    ) as [number, number];

    // Memo
    // Arry of date time strings
    const times = useMemo(() => data.timeSteps, [data]);

    // Handlers
    // Get date time string by index
    const outputFunction = useCallback((step: number) => times[step], [times]);
    // Update time range in redux. When the time aggregation is off,
    // only the upper bound of the range will be used in computing the plot data
    const onChange = useCallback(
        (range: unknown) =>
            dispatch(updateTimeIndexRange(range as [number, number])),
        [dispatch]
    );

    // Render
    return (
        <Root className={classes.root}>
            {/* This only appears when time aggregation is on */}
            {timeAggregation !== "None" && (
                <NativeSelect
                    className={classes.selector}
                    id="time-start-selector"
                    label="Start"
                    value={Math.min(...timeIndexRange)}
                    onChange={(event) =>
                        onChange([
                            parseInt(event.target.value),
                            Math.max(
                                parseInt(event.target.value),
                                timeIndexRange[1]
                            ),
                        ])
                    }
                >
                    {/* Show the full list of date times */}
                    {times.map((time, index) => (
                        <option
                            key={`time-dropdown-start-${time}`}
                            value={index}
                        >
                            {time}
                        </option>
                    ))}
                </NativeSelect>
            )}
            {/* Slider that is easy to use when the number of time steps is not too large */}
            <div className={classes.slider}>
                <span>Time Steps</span>
                <EdsSlider
                    track={timeAggregation === "None" ? false : "normal"}
                    aria-labelledby="time-step-slider-label"
                    value={
                        timeAggregation === "None"
                            ? Math.max(...timeIndexRange)
                            : timeIndexRange.slice()
                    }
                    valueLabelDisplay="on"
                    onChange={(_, value) =>
                        onChange(
                            //If time aggregation is off, we only need to know the end date
                            timeAggregation === "None"
                                ? [0, value]
                                : // This is due to a feature (or a bug) in EdsSlider that the first
                                  //value in the range is not necessarily the lower bound
                                  [
                                      Math.min(...(value as number[])),
                                      Math.max(...(value as number[])),
                                  ]
                        )
                    }
                    min={0}
                    max={times.length - 1}
                    step={1}
                    marks={true}
                    valueLabelFormat={outputFunction}
                    classes={{
                        root: classes.root,
                        valueLabel: classes.valueLabel,
                    }}
                />
            </div>
            <NativeSelect
                className={classes.selector}
                id="time-end-selector"
                label={timeAggregation === "None" ? "Select Time" : "End"}
                value={Math.max(...timeIndexRange)}
                onChange={(event) =>
                    onChange([
                        timeAggregation === "None" ? 0 : timeIndexRange[0],
                        parseInt(event.target.value),
                    ])
                }
            >
                {/* If time aggregation is on, we only show the time steps that >= the start date */}
                {(timeAggregation === "None"
                    ? times
                    : times.filter(
                          (_, index) => index >= Math.min(...timeIndexRange)
                      )
                ).map((time, index) => (
                    <option
                        key={`time-dropdown-end-${time}`}
                        value={
                            index +
                            (timeAggregation === "None"
                                ? 0
                                : Math.min(...timeIndexRange))
                        }
                    >
                        {time}
                    </option>
                ))}
            </NativeSelect>
        </Root>
    );
});

TimeRangeSelector.displayName = "TimeRangeSelector";
export default TimeRangeSelector;
