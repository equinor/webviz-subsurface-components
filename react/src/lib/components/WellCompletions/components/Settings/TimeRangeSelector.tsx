import { NativeSelect } from "@equinor/eds-core-react";
import {
    createStyles,
    makeStyles,
    Slider,
    Theme,
    // eslint-disable-next-line prettier/prettier
    withStyles
} from "@material-ui/core";
import { isEqual } from "lodash";
import React, { useCallback, useContext, useMemo } from "react";
import { useDispatch, useSelector } from "react-redux";
import { updateTimeIndexRange } from "../../redux/actions";
import { WellCompletionsState } from "../../redux/store";
import { DataContext } from "../DataLoader";
const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            display: "flex",
            flexDirection: "row",
        },
        slider: {
            width: "200px",
            marginLeft: theme.spacing(4),
            marginRight: theme.spacing(4),
        },
        selector: {
            maxWidth: "130px",
        },
    })
);
const EdsSlider = withStyles({
    root: {
        color: "#007079",
    },
    valueLabel: {
        top: 22,
        "& *": {
            background: "transparent",
            color: "#000",
        },
    },
})(Slider);
/**
 * A React component for selecting time step(s) to display in the plot
 */
const TimeRangeSelector: React.FC = React.memo(() => {
    const classes = useStyles();
    // Direct access to the input data
    const data = useContext(DataContext);

    // Redux
    const dispatch = useDispatch();
    const timeAggregation = useSelector(
        (state: WellCompletionsState) => state.ui.timeAggregation
    );
    const timeIndexRange = useSelector(
        (state: WellCompletionsState) => state.ui.timeIndexRange,
        isEqual
    ) as [number, number];

    // Memo
    // Arry of date time strings
    const times = useMemo(() => data.timeSteps, [data]);

    // Handlers
    // Get date time string by index
    const outputFunction = useCallback((step: number) => times[step], [times]);
    // Update time range in redux. When the time aggregation is off,
    // only the upper bound of the range will be used in computing the plot data
    const onChange = useCallback(
        (range) => dispatch(updateTimeIndexRange(range)),
        [dispatch]
    );

    // Render
    return (
        <div className={classes.root}>
            {/* This only appears when time aggregation is on */}
            {timeAggregation !== "None" && (
                <NativeSelect
                    className={classes.selector}
                    id="time-start-selector"
                    label="Start"
                    value={Math.min(...timeIndexRange)}
                    onChange={(event) =>
                        onChange([
                            parseInt(event.target.value),
                            Math.max(
                                parseInt(event.target.value),
                                timeIndexRange[1]
                            ),
                        ])
                    }
                >
                    {/* Show the full list of date times */}
                    {times.map((time, index) => (
                        <option
                            key={`time-dropdown-start-${time}`}
                            value={index}
                        >
                            {time}
                        </option>
                    ))}
                </NativeSelect>
            )}
            {/* Slider that is easy to use when the number of time steps is not too large */}
            <div className={classes.slider}>
                <span>Time Steps</span>
                <EdsSlider
                    track={timeAggregation === "None" ? false : "normal"}
                    aria-labelledby="time-step-slider-label"
                    value={
                        timeAggregation === "None"
                            ? Math.max(...timeIndexRange)
                            : timeIndexRange.slice()
                    }
                    valueLabelDisplay="on"
                    onChange={(_, value) =>
                        onChange(
                            //If time aggregation is off, we only need to know the end date
                            timeAggregation === "None"
                                ? [0, value]
                                : // This is due to a feature (or a bug) in EdsSlider that the first
                                  //value in the range is not necessarily the lower bound
                                  [
                                      Math.min(...(value as number[])),
                                      Math.max(...(value as number[])),
                                  ]
                        )
                    }
                    min={0}
                    max={times.length - 1}
                    step={1}
                    marks={true}
                    valueLabelFormat={outputFunction}
                />
            </div>
            <NativeSelect
                className={classes.selector}
                id="time-end-selector"
                label={timeAggregation === "None" ? "Select Time" : "End"}
                value={Math.max(...timeIndexRange)}
                onChange={(event) =>
                    onChange([
                        timeAggregation === "None" ? 0 : timeIndexRange[0],
                        parseInt(event.target.value),
                    ])
                }
            >
                {/* If time aggregation is on, we only show the time steps that >= the start date */}
                {(timeAggregation === "None"
                    ? times
                    : times.filter(
                          (_, index) => index >= Math.min(...timeIndexRange)
                      )
                ).map((time, index) => (
                    <option
                        key={`time-dropdown-end-${time}`}
                        value={
                            index +
                            (timeAggregation === "None"
                                ? 0
                                : Math.min(...timeIndexRange))
                        }
                    >
                        {time}
                    </option>
                ))}
            </NativeSelect>
        </div>
    );
});

TimeRangeSelector.displayName = "TimeRangeSelector";
export default TimeRangeSelector;
