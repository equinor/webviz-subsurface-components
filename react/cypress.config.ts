import { defineConfig } from "cypress";
import getCompareSnapshotsPlugin from "cypress-image-diff-js/dist/plugin";

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config),
      on('before:browser:launch', (browser = {}, launchOptions) => {
        if (browser.family === 'chromium' && browser.name !== 'electron') {
          launchOptions.args.push('--start-fullscreen')
      
          return launchOptions
        }
        if (browser.name === 'electron') {
          launchOptions.preferences.fullscreen = true
      
          return launchOptions
        }
      })
    }
  },
});
