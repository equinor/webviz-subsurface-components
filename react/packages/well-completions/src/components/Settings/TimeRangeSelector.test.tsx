import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "jest-styled-components";
import React from "react";
import { testStore, Wrapper } from "../../test/TestWrapper";
import TimeRangeSelector from "./TimeRangeSelector";

describe("test time range selector", () => {
    it("snapshot test", () => {
        const { container } = render(
            Wrapper({ children: <TimeRangeSelector /> })
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("click to dispatch redux action", async () => {
        render(Wrapper({ children: <TimeRangeSelector /> }));
        userEvent.type(screen.getByRole("slider"), "{arrowright}");
        expect(testStore.dispatch).toHaveBeenCalledTimes(1);
        expect(testStore.dispatch).toHaveBeenNthCalledWith(1, {
            payload: [0, NaN],
            type: "ui/updateTimeIndexRange",
        });
    });
});
