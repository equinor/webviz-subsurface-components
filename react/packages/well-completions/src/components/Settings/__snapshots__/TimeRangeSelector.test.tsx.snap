// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test time range selector snapshot test 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  margin-left: 8px;
  margin-right: 8px;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
}

.c2 {
  margin: 0;
}

.c3 {
  border: none;
  border-radius: 0;
  box-shadow: inset 0 -1px 0 0 var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  padding-left: 8px;
  padding-top: 6px;
  padding-right: 8px;
  padding-bottom: 6px;
  margin: 0;
  color: var(--eds_text__static_icons__tertiary,rgba(111,111,111,1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.500em;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-align: left;
  padding-right: calc(8px *2 + 24px);
  display: block;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%236f6f6f' d='M7 9.5l5 5 5-5H7z'/%3E%3C/svg%3E"), linear-gradient( to bottom, var(--eds_ui_background__light,rgba(247,247,247,1)) 0%, var(--eds_ui_background__light,rgba(247,247,247,1)) 100% );
  background-repeat: no-repeat,repeat;
  background-position: right 8px top 50%;
  width: 100%;
}

.c3:active,
.c3:focus {
  box-shadow: none;
  outline: 2px solid var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  outline-offset: 0px;
}

.c3:disabled {
  color: var(--eds_interactive__disabled__text,rgba(190,190,190,1));
  background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill='%23bebebe' d='M7 9.5l5 5 5-5H7z'/%3E%3C/svg%3E"), linear-gradient( to bottom, var(--eds_ui_background__light,rgba(247,247,247,1)) 0%, var(--eds_ui_background__light,rgba(247,247,247,1)) 100% );
  cursor: not-allowed;
  box-shadow: none;
  outline: none;
}

.c3:disabled .arrow-icon {
  fill: red;
}

.c3:disabled:focus,
.c3:disabled:active {
  outline: none;
}

.c0 {
  min-width: 100px;
  width: 100%;
}

<div
  class="TimeRangeSelector-root css-1htfc7f"
>
  <div
    class="TimeRangeSelector-slider"
  >
    <span>
      Time Steps
    </span>
    <span
      class="MuiSlider-root TimeRangeSelector-root MuiSlider-trackFalse MuiSlider-colorPrimary MuiSlider-sizeMedium css-1xvshl5-MuiSlider-root"
    >
      <span
        class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
      />
      <span
        class="MuiSlider-track css-pm1ix7-MuiSlider-track"
        style="left: 0%; width: 100%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="0"
        style="left: 0%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="1"
        style="left: 11.11111111111111%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="2"
        style="left: 22.22222222222222%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="3"
        style="left: 33.333333333333336%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="4"
        style="left: 44.44444444444444%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="5"
        style="left: 55.55555555555556%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="6"
        style="left: 66.66666666666667%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="7"
        style="left: 77.77777777777777%;"
      />
      <span
        class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
        data-index="8"
        style="left: 88.88888888888889%;"
      />
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="9"
        style="left: 100%;"
      />
      <span
        class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
        data-index="0"
        style="left: 100%;"
      >
        <input
          aria-labelledby="time-step-slider-label"
          aria-orientation="horizontal"
          aria-valuemax="9"
          aria-valuemin="0"
          aria-valuenow="9"
          data-index="0"
          max="9"
          min="0"
          step="1"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="9"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-valueLabelOpen MuiSlider-valueLabel TimeRangeSelector-valueLabel css-nnid7-MuiSlider-valueLabel"
        >
          <span
            class="MuiSlider-valueLabelCircle"
          >
            <span
              class="MuiSlider-valueLabelLabel"
            >
              2002-10-01
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>
  <div
    class="c0 TimeRangeSelector-selector"
  >
    <label
      class="c1"
      for="time-end-selector"
    >
      <span
        class="c2"
      >
        Select Time
      </span>
    </label>
    <select
      class="c3"
      id="time-end-selector"
    >
      <option
        value="0"
      >
        2000-02-01
      </option>
      <option
        value="1"
      >
        2000-06-01
      </option>
      <option
        value="2"
      >
        2000-07-01
      </option>
      <option
        value="3"
      >
        2000-10-01
      </option>
      <option
        value="4"
      >
        2001-01-01
      </option>
      <option
        value="5"
      >
        2001-03-01
      </option>
      <option
        value="6"
      >
        2001-04-01
      </option>
      <option
        value="7"
      >
        2001-10-01
      </option>
      <option
        value="8"
      >
        2002-07-01
      </option>
      <option
        value="9"
      >
        2002-10-01
      </option>
    </select>
  </div>
</div>
`;
